### PR TITLE
Pull Request for #13: Migrate demo scripts to `/examples/` and docs to wiki

### PR DESCRIPTION
**Pull request by gentzkow**
@ShiqiYang2022 @Xingtong-Jiang In this update, I have made two main changes to the template:

1. I moved all of the example scripts out of the main directories to `/examples/`. This way someone who downloads the template doesn't need to worry about having Python / R / Stata installed to be able to run `run_all.py` successfully. It also means there aren't so many example files floating around -- while my intention was that people would delete these, I've observed that they sometimes get left in repositories cluttering things up.  

2. I migrated the documentation for using the template to the GentzkowLabTemplate [wiki](https://github.com/gentzkow/GentzkowLabTemplate/wiki) and updated the [README](https://github.com/gentzkow/GentzkowLabTemplate/blob/84c7265b2e4f14127f9a3130f9a33ce62cccccff/README.md) to be a template for a repository readme. 

I'd appreciate it if you could both try running the template "out of the box", look over the documentation, and then try copying over and running the Python / Stata / R / Latex example scripts following the instructions [here](https://github.com/gentzkow/GentzkowLabTemplate/wiki#examples).

### CODE REVIEW

**Comment by ShiqiYang2022** [comment]
Thanks @gentzkow for setting this up! Confirming we have received this and will start working on testing soon.

**Review by Xingtong-Jiang — APPROVED** [review]
@gentzkow The update looks great to me! Here are my test results:
- Successfully ran the template with `run_all.sh` out of the box.
- Ran the template with Python after downloading libraries: `pandas`, `numpy`, `statsmodels`, `matplotlib`.
- Ran the template with Stata after downloading the `estout` package.
- Ran the template with R after downloading the `tidyverse` and `stargazer` packages.
- Ran the template with LaTeX after modifying `lib/shell/run_latex.sh` as suggested (See [comment](https://github.com/gentzkow/GentzkowLabTemplate/pull/14#discussion_r1705013983) below).
 
In the script files, I fixed some typos in a40de5190e6084870689877d5f46eee708308817 and 7f8e56cee0e5c0e33a0f517c34294459b6c741da.

I also flagged some minor issues in the documentation. On the [wiki](https://github.com/gentzkow/GentzkowLabTemplate/wiki) page:
- In [Setup](https://github.com/gentzkow/GentzkowLabTemplate/wiki#setup),
	- It seems the repository needs to be set as a template repository (Settings → General → Template repository) before one can navigate to its [GitHub page](https://github.com/gentzkow/GentzkowLabTemplate) and find the `Use this template` option.
	- `check_setup.sh` is not in the root directory.
- In [Examples](https://github.com/gentzkow/GentzkowLabTemplate/wiki#examples),
	- The use of periods at the end of bullet points is inconsistent.
- In [Repository organization - Overview](https://github.com/gentzkow/GentzkowLabTemplate/wiki#overview-1),
	- "`README.txt`: This readme file." no longer applies.
	- "`check_setup.sh`" doesn't match the name in the root directory, "`setup.sh`".
	- In the line of "`run_all.sh`", there is a missing period at the end.
- In [Repository organization - Modules](https://github.com/gentzkow/GentzkowLabTemplate/wiki#modules),
	- "The default structure..." has a missing T in front.
	- The bullet points listed under "key principles" use periods inconsistently at the end.
- In [Repository organization - Inputs and outputs](https://github.com/gentzkow/GentzkowLabTemplate/wiki#inputs-and-outputs),
	- I read the sentence "the code is agnostic about the location of files in the local directory structure" twice to understand what "files" and "local directory structure" mean. I'm not sure if it helps to replace it with something like "the code is agnostic about the location of ... files within the module's directory structure".
	- The use of periods at the end of each bullet point is inconsistent.
- In [Repository organization - External dependencies](https://github.com/gentzkow/GentzkowLabTemplate/wiki#external-dependencies),
	- I assume that `user_env.sh` is referring to `local_env.sh`.

Let me know if there are other ways I can help with this update. Thanks!

**Review by ShiqiYang2022 — APPROVED** [review]
@gentzkow Thanks for restructuring our lab template! The revised version looks great to me.

I initialized the branch [`13-testing`](https://github.com/gentzkow/GentzkowLabTemplate/tree/13-testing), and pushed my (success) testing results for [Out-of-box](https://github.com/gentzkow/GentzkowLabTemplate/commit/cb344f1edd22246b8c41af69e327f0b2c24812c8), [Python](https://github.com/gentzkow/GentzkowLabTemplate/commit/2a40592e822e3ca2409df8a55268ee5387f2f866), [Stata](https://github.com/gentzkow/GentzkowLabTemplate/commit/93e8ca3a9c68069c151dfda64f54381bbc89dce8), [R](https://github.com/gentzkow/GentzkowLabTemplate/commit/3bb399e345309fd90a3391f107778dbe8f482f2c), [Latex](https://github.com/gentzkow/GentzkowLabTemplate/commit/0776f6e76d148f089629b19383753a55d43124ec), and [PPTX](https://github.com/gentzkow/GentzkowLabTemplate/commit/796c6437de75138714b5ba42645b676040034116) accordingly. 

I posted my suggestions for the revised repo in each conversation comments.

For `Wiki`, my comments for details overlaps with @Xingtong-Jiang. Aside from those, I have one general comment:

I like the idea moving example scripts out of the main directories to `/examples/`. I also think another main advantage for this, is to provide an "empty" version for users to run very conveniently out of the box. However, by doing this, we might pay less attention in dependencies set-up and package version control (`conda`) in this template. Shall we consider to add that as a section in `Wiki` (I think that's our previous plan but will create additional complexities)? Or we can just ask users to refer to [packages](https://github.com/gentzkow/template_archive?tab=readme-ov-file#adding-packages) and [conda setup](https://github.com/gentzkow/template_archive?tab=readme-ov-file#setup) sections in our archived template if needed?

File: examples/latex/README.md
Comment (gentzkow) [reply examples/latex/README.md:3]: Approved!

File: 1_data/make.sh
Comment (gentzkow) [reply 1_data/make.sh:12]: @ShiqiYang2022 I would like to keep this in all the make.sh scripts. The reason is that we should assume not all users will run `run_all.sh` and they may skip running `check_setup.sh` by hand. For example, someone might clone a repository and then just try to compile a single module (like `paper` or `slides`). In that case we want to set things up so `make.sh` will run successfully and produce nice error messages in the case where the setup is incorrect.

Comment (ShiqiYang2022) [reply 1_data/make.sh:12]: @gentzkow Thanks! Your https://github.com/gentzkow/GentzkowLabTemplate/pull/14#discussion_r1706077612 make sense to me. I think we converged here, so I am marking this thread as resolved.

File: lib/shell/run_latex.sh
Comment (gentzkow) [reply lib/shell/run_latex.sh]: @ShiqiYang2022 I committed a slightly different solution here that keeps ${outputdir} as a variable that can optionally be set by the user.

File: 1_data/make.sh
Comment (gentzkow) [reply 1_data/make.sh:12]: Per above, I'd rather not make this change

File: 2_analysis/make.sh
Comment (gentzkow) [reply 2_analysis/make.sh:12]: Per above, I'd rather not make this change

File: 3_slides/make.sh
Comment (gentzkow) [reply 3_slides/make.sh:12]: Per above, I'd rather not make this change

File: 4_paper/make.sh
Comment (gentzkow) [reply 4_paper/make.sh:12]: Per above, I'd rather not make this change

**Comment by gentzkow** [comment]
@Xingtong-Jiang Thanks for your comments [here](https://github.com/gentzkow/GentzkowLabTemplate/pull/14#pullrequestreview-2220396517). I updated the wiki to reflect them.

I made the change to make it a template repository. 

@ShiqiYang2022 @Xingtong-Jiang I went through your proposed commits and accepted some, modified, some, and rejected some. Feel free to go through the comments above and resolve them if you agree.

**Comment by gentzkow** [comment]
@ShiqiYang2022 Also, I would propose we get rid of `setup.sh`. My idea is that `check_setup.sh` can perform the same function. If you think it's nicer to have it at the top level to make it more prominent, I'd be fine moving `check_setup.sh` back out of `lib/shell/` to the root of the repo.

File: lib/shell/run_latex.sh
Comment (ShiqiYang2022) [reply lib/shell/run_latex.sh]: @gentzkow Thanks! c8940982568bfe4ba1e09727275df34dab523904 is definitely a better solution. I am marking this as resolved.

Comment (gentzkow) [reply lib/shell/run_latex.sh]: Thanks! I actually haven't tested this so can you just confirm that it runs correctly?

**Comment by ShiqiYang2022** [comment]
@gentzkow Thanks! Re your https://github.com/gentzkow/GentzkowLabTemplate/pull/14#issuecomment-2272122797: I agree that we don't need `setup.sh`. I've also noticed that when I reviewed the repo. I went ahead and removed in [1888b97](https://github.com/gentzkow/GentzkowLabTemplate/pull/14/commits/1888b97f9f08a551cf4cab931a8e918b78feaacf).

File: lib/shell/run_latex.sh
Comment (ShiqiYang2022) [reply lib/shell/run_latex.sh]: @gentzkow Thanks and yes! I've confirm it runs well on my end, both with an output path and without an output path.

File: examples/inputs_for_examples/README.md
Comment (ShiqiYang2022) [reply examples/inputs_for_examples/README.md:1]: I went ahead and solved this by reorganizing the files in `/examples/inputs_for_examples/` in 906327a and 411116b, the current directory can be found [here](https://github.com/gentzkow/GentzkowLabTemplate/tree/411116b6f6ad9fc0f94d893c9cb345ebd4c7d91a/examples/inputs_outputs_for_examples).

@gentzkow I think (?) you probably missed this thread, so I went ahead and implement myself. Feel free to twist my edits if you disagree!

Comment (gentzkow) [reply examples/inputs_for_examples/README.md:1]: I agree!

Comment (ShiqiYang2022) [reply examples/inputs_for_examples/README.md:1]: Awesome! Resolving the thread!

**Comment by gentzkow** [comment]
@ShiqiYang2022 @Xingtong-Jiang 

I made a few more tweaks here
* I added more extensions to `.gitignore` so we're ignoring all the auxiliary files I know of produced by LaTex builds
* I set up the default `.gitattributes` for Git LFS and added code to `check_setup.sh` to verify the Git LFS is installed.
* I revised the `make.sh` scripts so they are all identical; I want to preserve this property so we don't have to keep updating them separately every time there's a change
* I tweaked the Stata scripts so they don't produce graphs with blue backgrounds
* I reorganized `inputs_for_examples` again to simplify; I think we want to have just the `.csv` file plus the other intermediate outputs in case someone wants to run the paper or slides without setting up analysis. We don't need the intermediate data files because if someone is going to set up R / Stata / Python they can set up both data and analysis.
* I revised the `README.md` for the LaTex example directory. I plan to revise the ones in the other directories similarly but I haven't had a chance. I think we should wait for that until this PR is merged then I can do it in another issue.

I would appreciate it if you guys could do one more quick round of testing to make sure I haven't broken anything.

Next steps for me/us after this issue is closed
1. Revise those readmes
2. Work more on how to robustly require Git LFS and also create an easy option for people who want to use the repo without Git LFS
3. Consider whether there's a good way to get rid of the redundancy of having multiple copies of `make.sh`. One would be to restore `setup.sh` and have it populate the various directories with copies of a single `make.sh` script. We could consider adding to this code to set up the LaTex, Stata, etc. examples and prompting the user to input which of these they would like.

**Comment by ShiqiYang2022** [comment]
@gentzkow 

I initialized a test branch [`13-testing-0812`](https://github.com/gentzkow/GentzkowLabTemplate/tree/13-testing-0812), did an another round of test on that branch and pushed the outputs there. Everything looks good to me, I made some small tweaks in ef9b324. Feel free to revert changes if you disagree.

I agree with your roadmap per https://github.com/gentzkow/GentzkowLabTemplate/pull/14#issuecomment-2285161889, and I and @Xingtong-Jiang are willing to contribute. Shall we open the issues and allocate among ourselves after we closed this?

Thanks!

**Comment by gentzkow** [comment]
@ShiqiYang2022 Looks good, modulo one tiny comment I made.

File: lib/shell/check_setup.sh
Comment (ShiqiYang2022) [inline lib/shell/check_setup.sh:36]:
```suggestion
echo "Setup check complete."
```

**Pull request merged**
